spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -270,7 +270,8 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		String taskDeploymentId = null;
 		try {
 			TaskLauncher taskLauncher = findTaskLauncher(platformName);
-			if(!isAppDeploymentSame(previousManifest, taskManifest)) {
+			Launcher launcher = this.launcherRepository.findByName(platformName);
```

Comment:
Why do we need to get the launcher again (we just got it the line above)?

Reply:
Different launcher ;-).